### ISSUE TITLE
Alternator fails to reject oversized requests if not using Content-Length

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#8196]
Alternator places a limit on the request size, using Seastar's HTTP server feature of `content_length_limit`.

What the Seastar code does is just to check the `Content-Length` header... But a long request can be sent without such a header - using chunked-encoding or if the connection is to be closed after the request - even without any pre-specified length at all. Such requests are currently allowed at any length, which is wrong.

We could add support in Seastar for this case as well: We have now a `content_stream` stream, and when the N+1th byte is read from the stream, or perhaps earlier when the next chunked-encoding chunk is known to go beyond position N, the read from the stream could throw an exception. This exception would cause the request handler to abort what it's doing. If the first Content-Length or chunk already go beyond N, we don't have to run the handler at all (this is what happens today).

Because the `content_length_limit` is a Seastar feature, we will need to open a corresponding issue for Seastar. We actually have two choices on what to do in Seastar:
1. Complete the Seastar feature - perhaps in the way suggested above.
2. Drop the incomplete feature from Seastar - and instead do the checking in Scylla: the JSON-parsing code already tracks the length of the parsing, and can throw when reaching the limit. Note that in Scylla, we can't report the error as soon as we see the Content-Length or chunk length and we can only report it once the Nth byte was actually read by the handler. I'm not sure this is bad.

**Comment by avikivity** [comment]
I don't think it merits a seastar feature. You should read from the content stream, and if you decide it's too long, close the stream and abort the request with an error. This way the condition (too long content) isn't hard-coded into Seastar but is left to the application.

**Comment by nyh** [comment]
For reasons explained above, I want to fix this issue in Alternator, not in Seastar. The fix should probably be in `server::handle_api_request(`)`, which reads the entire request into non-contiguous memory - unfortunately we must read the entire request before we can verify the validity of the signature - that depends on the content of the entire request. 

The current implementation simply calls util::read_entire_stream() to read the entire content of the request. Unfortunately this is a Seastar function so we can't easily modify it, but we can easily duplicate it (it's a very short function), and add to it a limitation that once we've passed a certain length, the read is aborted with an appropriate exception.
